merlin/util/dataportal.py

```diff
@@ -341,11 +342,39 @@ def get_sibling_with_extension(self, newExtension: str):
             self._exchange_extension(newExtension), self._client)
 
     def read_as_text(self):
-        return self._fileHandle.download_as_string().decode('utf-8')
+        """
+        Attempts to read a file from bucket as text, it if encounters a timeout
+        exception it reattempts after sleeping for exponentially increasing
+        delays, up to a delay of about 4 minutes
+        """
+        backoffSeries= [1, 2, 4, 8, 16, 32, 64, 128, 256]
+        for sleepDuration in backoffSeries:
+            try:
+                file = self._fileHandle.download_as_string().decode('utf-8')
+                return file
+            except (exceptions.GatewayTimeout, exceptions.ServiceUnavailable):
+                if sleepDuration == backoffSeries[-1]:
+                    raise
+                else:
+                    sleep(sleepDuration)
 
     def read_file_bytes(self, startByte, endByte):
-        return self._fileHandle.download_as_string(
-            start=startByte, end=endByte-1)
+        """
+        Attempts to read a file from bucket as bytes, it if encounters a timeout
+        exception it reattempts after sleeping for exponentially increasing
+        delays, up to a delay of about 4 minutes
+        """
+        backoffSeries= [1, 2, 4, 8, 16, 32, 64, 128, 256]
+        for sleepDuration in backoffSeries:
+            try:
+                file = self._fileHandle.download_as_string(start=startByte,
```

Comment:
This is very redundant with above. Can we move the retry logic to retry to its own function that we don't have replicated code?